app/routes/steps/step1.tsx

```diff
@@ -0,0 +1,49 @@
+import { Button } from "@digitalservice4germany/digital-service-library";
+import {
+  useLoaderData,
+  Form,
+  ActionFunction,
+  LoaderFunction,
+  redirect,
+} from "remix";
+import { getFormDataCookie, getFormDataCookieResponseHeader } from "~/cookies";
+
+export const loader: LoaderFunction = async ({ request }) => {
+  const formData = await getFormDataCookie(request);
+  return [formData];
+};
+
+export const action: ActionFunction = async ({ request }) => {
+  const formData: FormData = await request.formData();
+  const street: string = "" + formData.get("street");
+
+  const cookie: Record<string, string> = await getFormDataCookie(request);
+  cookie["street"] = street;
+
+  const responseHeader: Record<string, string> =
+    await getFormDataCookieResponseHeader(cookie);
+  return redirect("/steps/step2", {
+    headers: responseHeader,
+  });
+};
+
+export default function Step1() {
+  const formData = useLoaderData()[0]; // TODO do I have to map here?
```

Comment:
@frederike-ramin I think you can freely choose what to return from the loader function. At the moment you return an array with on item. Alternatively you could return `formData` directly.

```
return formData;
....
const formData = useLoaderData();
```

Reply:
Oh my, nice! I blindly followed the intro. This is great! 👍